### PR TITLE
Switch to buildx + qemu-static for building and deploying

### PR DESCRIPTION
**Pull request by andreabolognani**
As suggested by @rmohr, we don't actually need to have multiple workers running builds in parallel and coordinating for creating the manifest: we can take advantage of qemu-static instead, and use a single amd64 worker for everything.

In the process, we switch to using the buildx Docker plugin for builds.

### CODE REVIEW

File: hack/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+sudo cp hack/docker-daemon.json /etc/docker/daemon.json
+sudo systemctl restart docker
+
+# Install the buildx Docker plugin
+DOCKER_BUILDKIT=1 docker build --platform=local -o . git://github.com/docker/buildx
+mkdir -p ~/.docker/cli-plugins
+mv buildx ~/.docker/cli-plugins/docker-buildx
+
+# Set up the cross-architecture build environment
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```
Comment (rmohr) [inline hack/prepare.sh:27]: I understand this line but I don't understand how builds can work afterwards.

Are you somewhere copying the qemu-static binary into the container? If not, how can e.g. the arm64 container reach the corresponding qemu-static?

Comment (andreabolognani) [reply hack/prepare.sh:27]: Basically when `-p yes` is used the qemu-static binaries are loaded into memory and no longer need to be accessed from the filesystem, which means that any container from any architecture can be used transparently from that moment forward. See https://github.com/multiarch/qemu-user-static#multiarch-compatible-images-deprecated.

Comment (rmohr) [reply hack/prepare.sh:27]: Nice. I missed that flag.

Comment (andreabolognani) [reply hack/prepare.sh:27]: @rmohr anything else I need to do to get this in?

**Review by rmohr — APPROVED** [review]

**Pull request merged**